### PR TITLE
[TASK] Remove api's dependency on response-handler

### PR DESCRIPTION
**Pull request by clark800**
'request' parameter to API calls will be split into multiple parameters in a future pull request. 'wrapper' function in router is temporary, there will be a custom wrapper for each function in the server.

### CODE REVIEW

File: api/orders.js [hunk 1]
```diff
@@ -155,10 +152,8 @@ function getOrders(request, response, next) {
  *  @query
  *  @param {String "true"|"false"} request.query.validated    - used to force request to wait until rippled has finished validating the submitted transaction
  *
- *  @param {Express.js Response} response
```
Comment (lumberj) [inline api/orders.js]: Maybe keep the jsdoc block updated

```javascript
* @callback requestCallback
* @param {Error} [err]
* @param {Object} responseObject

* @param {requestCallback} cb
```


(http://usejsdoc.org/tags-param.html)

File: server/express_app.js [hunk 2]
```diff
@@ -36,7 +36,6 @@ if (config.get('debug')) {
 }
 
 app.use('/v' + version.getApiVersion(), router);
-app.use('/', (new express.Router()).get('/', router.generateIndexPage));
```
Comment (lumberj) [inline server/express_app.js]: Why is this removed?

File: server/router.js [hunk 3]
```diff
@@ -5,9 +6,42 @@ const errors    = require('../api').errors;
...
+    '/wallet/new': api.wallet.generate,
+    '/accounts/:account/payments': api.payments.getAccountPayments,
+    '/accounts/:account/payments/:identifier': api.payments.get,
+    '/accounts/:account/payments/paths/:destination_account/:destination_amount_string': api.payments.getPathFind,
+    '/accounts/:account/orders': api.orders.getOrders,
+    '/accounts/:account/order_book/:base/:counter': api.orders.getOrderBook,
+    '/accounts/:account/orders/:identifier': api.orders.getOrder,
+    '/accounts/:account/notifications': api.notifications.getNotification,
+    '/accounts/:account/notifications/:identifier':
+      api.notifications.getNotification,
```
Comment (lumberj) [inline server/router.js]: Is this meant to be on a separate line?

File: server/router.js [hunk 4]
```diff
@@ -5,9 +6,42 @@ const errors    = require('../api').errors;
 const respond   = require('./response-handler.js');
 const version   = require('./version.js');
 
+const ROUTE_MAP = {
```
Comment (lumberj) [inline server/router.js]: I think you're missing the root path `/` from here

File: api/orders.js (hunk 1 above)
Comment (lumberj) [inline api/orders.js]: Or, since this is in flux, maybe do this later.

File: server/express_app.js (hunk 2 above)
Comment (clark800) [inline server/express_app.js]: It's redundant isn't it?

File: server/router.js (hunk 3 above)
Comment (clark800) [inline server/router.js]: Yeah, over the 80 character limit.

File: server/router.js (hunk 4 above)
Comment (clark800) [inline server/router.js]: Intentionally registered separately because it uses a different calling syntax

File: api/orders.js (hunk 1 above)
Comment (clark800) [inline api/orders.js]: Yeah, that's what I was thinking, the rest of the parameters are going to change in the next pull request.

File: server/router.js (hunk 3 above)
Comment (lumberj) [inline server/router.js]: I usually do 120 characters. Also, line 18 is definitely longer

**Comment by lumberj** [comment]
LGTM

File: server/express_app.js (hunk 2 above)
Comment (geertweening) [inline server/express_app.js]: It makes it so that `/v1` as well as `/` generate the index page. Otherwise `/` returns "Not Found"

Comment (clark800) [inline server/express_app.js]: I added this: https://github.com/ripple/ripple-rest/pull/297/files#diff-79279ffabc15c88f1859a5273ec21eccR42

File: server/router.js [hunk 5]
```diff
@@ -5,9 +6,42 @@ const errors    = require('../api').errors;
...
+};
+
 var router = new express.Router();
+router.get('/', generateIndexPage);
 
-router.generateIndexPage = function(req, res) {
+function generateIndexPage(req, res) {
   var url_base = '/v' + version.getApiVersion();
 
   res.json({
```
Comment (geertweening) [inline server/router.js]: This isn't part of your changes, but since I'm not noticing it while looking at the diff. Responses should only be handled by the response handler.

**Comment by geertweening** [comment]
Besides open comments, LGTM
